src/linear_solver_routines/iterative_hessian_sketch.jl

```diff
@@ -0,0 +1,72 @@
+# This file is part of RLinearAlgebra.jl
+# 1. Specifies type
+# 2. Implements rsubsolve function
+
+"""
+    IterativeHessianSketch <: LinSysSolveRoutine
+
+A mutable structure that represents the Iterative Hessian Sketch Method.
+
+See Mert Pilanci and Martin J. Wainwright. "Iterative Hessian Sketch: 
+    Fast and Accurate Solution Approximation for Constrained Least-Squares."
+    Journal of Machine Learning Research, 17(53), 1–38.
+    http://jmlr.org/papers/v17/14-460.html
+
+# Fields
+
+- `A::AbstractArray` is the coefficient matrix for the linear system. Required as we need to compute the full residual.
+- `b::AbstractVector` is the constant term (i.e., the linear system is Ax=b).
```

Comment:
If you are noting that the matrix `A` is required to compute the residual the same note should be made for the constant vector.